frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-query-result/experiment-query-result.component.ts

```diff
@@ -32,23 +58,165 @@ export class ExperimentQueryResultComponent implements OnInit, OnDestroy {
+              let levelName = level.name;
+              // collect level names in 2 list
+              decisionPoint.factors.length >= 2 ? decisionPointIndex = factorIndex : decisionPointIndex = decisionpointIndex;
+              decisionPointIndex === 0 ? resultData1.push(levelName) : resultData2.push(levelName)
+            });
+          });
+        });
+
+        // factor 1 with factor 2
+        emptySeries1 = this.prepareEmptySeriesInteractionGraphData(resultData1, resultData2);
+        // factor 2 with factor 1
+        emptySeries2 = this.prepareEmptySeriesInteractionGraphData(resultData2, resultData1);
+        
+        // fill the result values for each query:
+        let resData1 = emptySeries1;
+        let resData2 = emptySeries2;
+        res.interactionEffect.map((data) => {
+          // levels of the condition:
+          let levels = this.getLevels(data.conditionId);
+          this.populateLineChartSeries(resData1, data, levels, 0);
+          this.populateLineChartSeries(resData2, data, levels, 1);
+        });
+        this.interactionEffectQueryFactorResults1 = {
+          ...this.interactionEffectQueryFactorResults1,
+          [res.id]: resData1,
         };
+        this.interactionEffectQueryFactorResults2 = {
+          ...this.interactionEffectQueryFactorResults2,
+          [res.id]: resData2,
+        };
+      }
+    });
+  }
+  
+  prepareEmptySeriesInteractionGraphData(resultData1, resultData2) {
+    let emptySeries = [];
+    resultData1.map((level1) => {
+      let series = [];
+      resultData2.map((level2) => {
+        series.push({
+            name: level2,
+            value: 0,
+            participantsLogged: 0
+        });
+      });
+      emptySeries.push({
+        name: level1,
+        series: series,
+        dot: true
+      });
+    });
+    return emptySeries;
+  }
+
+  populateLineChartSeries(resData: any[], data, levels, factorNumber: number) {
+    resData.map((resData) => {
+      if (resData.name === levels[factorNumber].level.name) {
+        return resData.series.map((level) => {
+          let alternateFactorNumber = factorNumber === 0 ? 1 : 0;
+          if (level.name === levels[alternateFactorNumber].level.name) {
+            level.value = Math.round(Number(data.result) * 100)/ 100;
+            level.participantsLogged = Number(data.participantsLogged);
+          }
+        });
+      }
+    });
+  }
+  
+  setMaxLevelsCount() {
+    this.experiment.partitions.map((decisionPoint, decisionPointIndex) => {
+      decisionPoint.factors.map((factor, factorIndex) => {
+        let levelCount = factor.levels.length;
+        if (levelCount > this.maxLevelCount) {
+          this.maxLevelCount = levelCount;
+        }
       });
     });
   }
 
+  setConditionCount() {
+    this.maxLevelCount = this.experiment.conditions.length;
+  }
+
+  getFactorIndex(levelId: any) {
+    let decisionPointIndex;
+    this.experiment.partitions.map((decisionPoint, decisionpointIndex) => {
```

Comment:
all of these maps that don't return anything ought to be `forEach` as they don't return anything